### PR TITLE
2290: Upgrade dependencies

### PR DESCRIPTION
**Pull request by sarahsporck**
### Short description

Upgrade dependencies.

### Proposed changes

- Update all trivial yarn dependencies
- Update fastlane
- Remove react-styled-flexboxgrid, which was carried with the project [for quite some time](https://github.com/digitalfabrik/integreat-webapp/commit/221e6421593babe56b90529b20cb333beb74af4a) even though now unused

### Side effects

- Basically anything can happen with dependency updates, therefore test thoroughly.
- Excluded Updates (because non trivial) or excluded major version updates:
  - Excluded styled components due to https://github.com/styled-components/styled-components/issues/4112
  - Excluded metro-resolver as there have been a few breaking changes, which also affect our code base. (#2402)
  - Excluded https://github.com/birdofpreyru/react-native-static-server (#2389)
  - Excluded react & react-native
  - Excluded i18next & react-i18next (#2014)
  - Excluded firebase as it would be incompatible with flipper on ios (#2403)
  - Excluded decamelize as it is now an ESM package (#2404)

### Resolved issues

Fixes: #2290

---

### CODE REVIEW

File: native/android/gradle.properties [hunk 1]
```diff
@@ -25,10 +25,7 @@ android.useAndroidX=true
 android.enableJetifier=true
 
 # Version of flipper SDK to use with React Native
-FLIPPER_VERSION=0.176.0
-
-# Overwrite minSdkVersion of react-native-static-server
-RNStaticServer_minSdkVersion=21
```
Comment (steffenkleinle) [inline native/android/gradle.properties:31]: Why did you remove this?

File: native/src/components/StaticServerProvider.tsx [hunk 2]
```diff
@@ -18,17 +18,18 @@ const staticServer = new StaticServer(SERVER_PORT, SERVER_PATH, {
 export const StaticServerContext = createContext('')
 
 const StaticServerProvider = ({ children }: StaticServerProps): ReactElement | null => {
-  const [resourceCacheUrl, setResourceCacheUrl] = useState<string | null>(null)
+  const [resourceCacheUrl, setResourceCacheUrl] = useState<string>('')
```
Comment (steffenkleinle) [inline native/src/components/StaticServerProvider.tsx:21]: Why these changes?

**Review by LeandraH — APPROVED** [review]
Looks good, tested on Firefox and real iOS

**Review by steffenkleinle — APPROVED** [review]
Tested on firefox and android.

I found one issue: It seems like showing images offline doesn't work anymore. Steps to reproduce:
- Go to testumgebung > de > Test 4
- Navigate back to testumgebung > de
- Disable internet
- Go back to Test 4
- No image

With the version in the store this works for me.

File: native/package.json [hunk 3]
```diff
@@ -50,89 +50,88 @@
...
+    "react-native-screens": "^3.24.0",
+    "react-native-svg": "^13.13.0",
     "react-native-system-setting": "^1.7.6",
-    "react-native-url-polyfill": "^1.3.0",
-    "react-native-vector-icons": "^9.2.0",
-    "react-native-webview": "^13.2.2",
-    "react-navigation-header-buttons": "^10.0.0",
-    "styled-components": "^5.3.5",
+    "react-native-url-polyfill": "^2.0.0",
+    "react-native-vector-icons": "^10.0.0",
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply native/package.json:95]: Necessary as long as we use react-native-elements. Will be removed in #2331.

**Review by steffenkleinle — CHANGES_REQUESTED** [review]
Wrongly approved, sorry

File: native/package.json [hunk 4]
```diff
@@ -50,89 +50,88 @@
...
-    "react-native-pdf": "^6.6.2",
-    "react-native-permissions": "^3.4.0",
-    "react-native-reanimated": "^2.14.4",
-    "react-native-safe-area-context": "^4.5.1",
-    "react-native-screens": "^3.19.0",
-    "react-native-svg": "^13.9.0",
+    "react-native-localize": "^3.0.2",
+    "react-native-pdf": "^6.7.1",
+    "react-native-permissions": "^3.8.4",
+    "react-native-reanimated": "^3.4.2",
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply native/package.json:89]: I think this was necessary for react-native-screens/reactnavigation.

File: native/package.json [hunk 5]
```diff
@@ -50,89 +50,88 @@
...
+    "@react-navigation/native": "^6.1.7",
+    "@react-navigation/stack": "^6.3.17",
+    "@sentry/react-native": "^5.9.1",
+    "@sentry/types": "^7.65.0",
     "accordion-collapse-react-native": "^1.1.1",
     "api-client": "0.0.1",
     "build-configs": "0.0.1",
-    "events": "^3.2.0",
-    "htmlparser2": "^7.2.0",
+    "events": "^3.3.0",
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply native/package.json:69]: This is (or at least was) required by htmlparser2, but I could not find this being required anymore. So maybe we can actually remove it?

**Comment by steffenkleinle** [comment]
Could you perhaps remove outdated required dependencies docs? Also react-native-upgrades docs could be removed bc we don't update them anymore.

File: native/android/gradle.properties (hunk 1 above)
Comment (sarahsporck) [reply native/android/gradle.properties:31]: we don't use react-native-static-server anymore and the new library doesn't seem to use this constant either: https://github.com/search?q=repo%3Abirdofpreyru%2Freact-native-static-server%20RNStaticServer_minSdkVersion&type=code

Comment (steffenkleinle) [reply native/android/gradle.properties:31]: Ahh good find!

File: native/src/routes/ExternalOffer.tsx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import { fromPairs } from 'lodash'
-import { stringify } from 'query-string'
+import queryString from 'query-string'
```
[the remark this answers is not in the document]
Comment (sarahsporck) [reply native/src/routes/ExternalOffer.tsx:2]: stringify is not exported directly here. Also this is not importing the whole package, but just the default export. Importing a whole package normally looks like this: `import * as queryString from 'query-string'` :)

[the remark this answers is not in the document]
Comment (LeandraH) [reply native/src/routes/ExternalOffer.tsx:2]: Alright, same question, why did you switch to the default export?

File: native/src/components/StaticServerProvider.tsx (hunk 2 above)
Comment (sarahsporck) [reply native/src/components/StaticServerProvider.tsx]: Ahh. These were a remainder of when i first tried update the static server package. This was also the reason why offline capability for images did not work. Reverting this fixed that.

**Review by steffenkleinle — APPROVED** [review]
Not tested again

File: native/src/routes/ExternalOffer.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (sarahsporck) [reply native/src/routes/ExternalOffer.tsx:2]: As I said stringify is not exported anymore, you can only import it using the default export: https://github.com/sindresorhus/query-string/blob/main/index.d.ts

[the remark this answers is not in the document]
Comment (LeandraH) [reply native/src/routes/ExternalOffer.tsx:2]: Aaaaah, cool, thanks for the explanation :)
